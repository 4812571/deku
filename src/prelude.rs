/*! Crate prelude

[What is a prelude?](https://doc.rust-lang.org/std/prelude/)
*/
pub use crate::{
    error::DekuError, DekuContainerRead, DekuContainerWrite, DekuRead, DekuUpdate, DekuWrite,
};
pub use bitvec::{
    order::BitOrder, order::Lsb0, order::Msb0, slice::BitSlice, vec::BitVec, view::BitView,
};
