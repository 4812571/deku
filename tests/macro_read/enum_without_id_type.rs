use deku::prelude::*;

#[derive(DekuRead)]
enum Test {}

fn main() {}
