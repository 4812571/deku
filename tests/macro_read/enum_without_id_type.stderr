error: `id_type` must be specified on enum
 --> $DIR/enum_without_id_type.rs:4:6
  |
4 | enum Test {}
  |      ^^^^
